`runner install` panics instead of returning an error message

**Describe the bug**
`runner install` panics instead of returning an error message when the specified image to use for the runner does not exist

**Steps to Reproduce**

Note that I typo'd the `docker-runner-image` so this should have failed; the issue is that I got a panic instead of a failure message.

```
❯ waypoint runner install \
  -platform=docker \
  -server-addr=api.hashicorp.cloud:443 \
  -docker-runner-image=ghcr.io/hashicorp/waypoint/alpha.
✓ Finished connecting to: api.hashicorp.cloud:443
❌ Installing runner...
❌ Initializing Docker client...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x105ac1728]

goroutine 1 [running]:
github.com/docker/distribution/reference.FamiliarString({0x0, 0x0})
	/home/circleci/go/pkg/mod/github.com/docker/distribution@v2.8.0+incompatible/reference/helpers.go:31 +0x88
github.com/hashicorp/waypoint/internal/runnerinstall.(*DockerRunnerInstaller).Install(0x14000b9bc80, {0x10817da00, 0x14000e52700}, 0x140009125a0)
	/home/circleci/project/waypoint/internal/runnerinstall/docker.go:48 +0x1b0
github.com/hashicorp/waypoint/internal/cli.(*RunnerInstallCommand).Run(0x14000b9ce40, {0x140001cf330, 0x3, 0x5})
	/home/circleci/project/waypoint/internal/cli/runner_install.go:206 +0x77c
github.com/mitchellh/cli.(*CLI).Run(0x14000e60000)
	/home/circleci/go/pkg/mod/github.com/mitchellh/cli@v1.1.2/cli.go:262 +0x5c4
github.com/hashicorp/waypoint/internal/cli.Main({0x14000122120, 0x6, 0x6})
	/home/circleci/project/waypoint/internal/cli/main.go:127 +0x5cc
main.main()
	/home/circleci/project/waypoint/cmd/waypoint/main.go:14 +0xb4
```

**Expected behavior**
I expect an informative error message telling me what I need to fix on my inputs, rather than a panic.

**Waypoint Platform Versions**
Additional version and platform information to help triage the issue if
applicable:

* Waypoint CLI Version: 0.9.0
* Waypoint Server Platform and Version: docker, 0.8.0

--- a/internal/runnerinstall/docker.go
+++ b/internal/runnerinstall/docker.go
@@ -41,6 +41,9 @@ func (i *DockerRunnerInstaller) Install(ctx context.Context, opts *InstallOpts) 
 
 	runnerImage := i.Config.RunnerImage
 	imageRef, err := reference.ParseNormalizedNamed(runnerImage)
+	if err != nil {
+		return err
+	}
 
 	imageList, err := cli.ImageList(ctx, types.ImageListOptions{
 		Filters: filters.NewArgs(filters.KeyValuePair{

